### PR TITLE
prov/util, verbs: fixes for counter wait and trywait.

### LINKED ISSUE #5109: When using rxm/verbs when fi_cntr_wait() is called the receipt of EQ events is not causing progress
@a-ilango - I see that the EQ FD is added to the counter wait object. What I'm seeing is that when fi_cntr_wait() is called, that in the call util_wait_fd_run(), fi_epoll_wait() returns 1. The loop then continues and the call wait->util_wait.wait_try() returns 0, so the code then calls fi_epoll_wait() again (which returns 1 again). This repeats until the timeout occurs.

### PR DESCRIPTION
**Pull request by a-ilango**
Fixes #5109

### CODE REVIEW

**Comment by shefty** [comment]
@a-ilango - can you please rebase over master and re-push?

**Comment by a-ilango** [comment]
Synced with upstream changes. Btw, I see a failure with tcp provider with commit 344e422. It seems the `SERVER_RECV_CONNREQ` event is not being written into the EQ. @vkrishna

**Comment by a-ilango** [comment]
@swelch Thanks for testing!

**Comment by vkrishna** [comment]
I am checking the tcp failures

File: prov/util/src/util_wait.c [hunk 1]
```diff
@@ -273,6 +273,9 @@ static int util_wait_fd_run(struct fid_wait *wait_fid, int timeout)
 			return -FI_ETIMEDOUT;
 
 		ret = fi_epoll_wait(wait->epoll_fd, ep_context, 1, timeout);
+		if (ret > 0)
+			return FI_SUCCESS;
```
Comment (vkrishna) [inline prov/util/src/util_wait.c]: The issue here is that eq_sread is returning even when there are no pending events yet in eq. 

For tcp case, wait's epoll only checks for events on socket level. Based on those we can't return eq_sread call, But that is happening here.

**Pull request merged**